### ISSUE TITLE
您好，请问MAC如何和GEO-transformer一起配准

### ISSUE THREAD

**Issue by 17854174130** [zhangxy0517/3D-Registration-with-Maximal-Cliques#21]
您好，请问有没有和GEO-transformer配合一起配准的源码？
向您请教下如何组合在GEO框架中？谢谢 ！

**Comment by 17854174130** [comment]
确实有很多问题，有点无从下手，希望可以得到您的帮助，感谢！

**Comment by zhangxy0517** [comment]
1、目前提供的python版本仅供演示用，如需实现完整功能请参照c++代码补充（linux registraion.cpp 的1055行至1227行）
2、替换 eval.py 的 evaluate registration部分

**Comment by 17854174130** [comment]
非常感谢您的回复：
        通过对文章和代码的阅读，我发现registraion.cpp 的1055行至1227行主要工作是搜索点云中的团结构，旨在加强其亲和力。可能是我的理解有误，将此处代码替换到eval.py 的 evaluate registration的工作使我无法理解，evaluate registration好像是评估配准工作。为了确定我正确理解您的意思，我将被替换的代码附在下面。
        为何要将此处代码替换到eval.py 的 evaluate registration？希望可以得到您的回复，感谢！
 
# 2. evaluate registration
            if args.method == 'lgr':
                estimated_transform = data_dict['estimated_transform']
            elif args.method == 'ransac':
                estimated_transform = registration_with_ransac_from_correspondences(  
                    src_corr_points,
                    ref_corr_points,
                    distance_threshold=cfg.ransac.distance_threshold,
                    ransac_n=cfg.ransac.num_points,
                    num_iterations=cfg.ransac.num_iterations,
                )
            elif args.method == 'svd':
                with torch.no_grad():
                    ref_corr_points = torch.from_numpy(ref_corr_points).cuda()
                    src_corr_points = torch.from_numpy(src_corr_points).cuda()
                    corr_scores = torch.from_numpy(corr_scores).cuda()
                    estimated_transform = weighted_procrustes( 
                        src_corr_points, ref_corr_points, corr_scores, return_transform=True
                    )
                    estimated_transform = estimated_transform.detach().cpu().numpy()
            else:
                raise ValueError(f'Unsupported registration method: {args.method}.')

            estimated_transforms.append(
                dict(
                    test_pair=[ref_frame, src_frame],
                    num_fragments=num_fragments,
                    transform=estimated_transform,
                )
            )

            if gt_indices[ref_frame, src_frame] != -1:

**Comment by zhangxy0517** [comment]
我的意思是MAC整体替换evaluate registration
